### PR TITLE
MMT-1984: add ability to create a metadata update proposal

### PR DESCRIPTION
**Pull request by charleshuang80**
* add link, modal, and controller actions for creating an update proposal
* move methods out of controller into model
* add tests

### CODE REVIEW

File: app/views/proposal/collections/show.html.erb [hunk 1]
```diff
@@ -17,12 +17,22 @@
     <section class="action wide-content">
       <div class="wide-content-inside">
         <span>
+          <%= link_to 'Create Update Proposal', '#update-request-modal', class: 'display-modal eui-btn--link bar-after' %>
```
Comment (charleshuang80) [inline app/views/proposal/collections/show.html.erb]: I checked with the PO, she prefers these actions to stay as links

File: spec/features/proposal/collection_draft_proposals/collection_draft_proposal_update_spec.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-describe 'Collection Draft Proposal Update', js: true do
+describe 'Collection Draft Proposal Update', reset_provider: true, js: true do
```
Comment (cgokey) [inline spec/features/proposal/collection_draft_proposals/collection_draft_proposal_update_spec.rb]: I mentioned this in of Ryan's pull requests, does adding/keeping js:true slow down the tests in your opinion?   I thought I noticed a significant difference one day when I was testing.

Comment (charleshuang80) [reply spec/features/proposal/collection_draft_proposals/collection_draft_proposal_update_spec.rb]: Yes it does. I think we should remove it where possible, but if your integration test requires things that the js is doing, then it is necessary. I have also found it necessary for capybara screenshots, but that can be used temporarily.

Comment (charleshuang80) [reply spec/features/proposal/collection_draft_proposals/collection_draft_proposal_update_spec.rb]: this test _might not_ need `js: true` but I'd rather not change it at the moment. with one of the future draft mmt tickets, I'll review if all the tests for proposals need it or not.

**Pull request merged**